clippy_lints/src/hash_collision.rs

```diff
@@ -0,0 +1,91 @@
+use clippy_utils::diagnostics::span_lint_and_note;
+use clippy_utils::last_path_segment;
+use clippy_utils::ty::is_type_diagnostic_item;
+use rustc_hir::ExprKind;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::declare_lint_pass;
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// When two items are inserted into a `HashMap` with the same key,
+    /// the second item will overwrite the first item.
+    ///
+    /// ### Why is this bad?
+    /// This can lead to data loss.
```

Comment:
```suggestion
    /// Warn when a `Hashmap` is set up with the
    /// same key appearing twice.
    ///
    /// ### Why is this bad?
    /// When two items are inserted into a `HashMap` with the same key,
    /// the second item will overwrite the first item.
    /// This can lead to data loss.
```